src/components/dialogs/DeleteMessageDialog.vue

```diff
@@ -0,0 +1,107 @@
+<template>
+  <q-card>
+    <q-card-section class="row items-center">
+      <q-avatar
+        icon="delete"
+        color="red"
+        text-color="white"
+      />
+      <span class="q-ml-sm">Are you sure you want to delete this message?</span>
+    </q-card-section>
+
+    <q-card-actions align="right">
+      <q-btn
+        flat
+        label="Cancel"
+        color="primary"
+        v-close-popup
+      />
+      <q-btn
+        flat
+        label="Delete"
+        color="primary"
+        v-close-popup
+        @click="deleteMessageBoth()"
+      />
+    </q-card-actions>
+  </q-card>
+</template>
+
+<script>
+import { mapActions, mapGetters } from 'vuex'
+import { caldIdFromParts } from '../../utils/wallet'
+import { sentTransactionFailureNotify } from '../../utils/notifications'
+
+export default {
+  props: ['address', 'index'],
+  methods: {
+    ...mapActions({
+      deleteMessage: 'chats/deleteMessage',
+      forwardTransaction: 'wallet/forwardTransaction',
+      getFee: 'wallet/getFee',
+      unfreezeUTXO: 'wallet/unfreezeUTXO'
+    }),
+    ...mapGetters({
+      getRelayClient: 'relayClient/getClient',
+      getElectrumClient: 'electrumHandler/getClient',
+      getToken: 'relayClient/getToken',
+      getMessage: 'chats/getMessage',
+      isUTXO: 'wallet/isUTXO',
+      getUTXO: 'wallet/getUTXO',
+      getChangeAddresses: 'wallet/getChangeAddresses',
+      getMyAddressStr: 'wallet/getMyAddressStr'
+    }),
+    async deleteMessageBoth () {
+      // TODO: Move this into wallet API
+      // TODO: More private
+
+      // Get utxos attached to message
+      const outpoints = this.getMessage()(this.address, this.index).outpoints
```

Comment:
What about stealth outpoints?  This is why I wanted a general UTXO interface even if most entries didn't have anything.